src/main.rs

```diff
@@ -1,10 +1,21 @@
 use std::env;
-
+use std::io::{self, Write, BufWriter};
+use std::process;
 fn main() {
-    let arguments = env::args().collect::<Vec<_>>();
-    
-    match arguments.len() {
-        1 => loop { println!("y") },
-        _ => loop { println!("{}", arguments[1]) },
+    {
+        // Move all locals into a scope, so that `process::exit()` is called after all destructors.
+        let output = env::args()
+            .skip(1)
+            .next()
+            .map(|x| x + "\n")
+            .unwrap_or("y\n".to_owned());
+        let stdout = io::stdout();
+        let mut buffered = BufWriter::new(stdout.lock());
+        loop {
```

Comment:
Actually, it should probably be:

```rust
while buffered.write_all(&output).is_ok() {}
```

(he's trying to repeatedly write the same string).